exchanges/gateio/gateio_wrapper.go

```diff
@@ -2172,21 +2173,110 @@ func (g *Gateio) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) e
 				minBaseAmount = math.Pow10(-int(pairsData[x].AmountPrecision))
 			}
 
-			limits = append(limits, order.MinMaxLevel{
-				Asset:                   a,
-				Pair:                    pair,
+			l = append(l, limits.MinMaxLevel{
+				Key:                     key.NewExchangePairAssetKey(g.Name, a, pair),
 				QuoteStepIncrementSize:  math.Pow10(-int(pairsData[x].Precision)),
 				AmountStepIncrementSize: math.Pow10(-int(pairsData[x].AmountPrecision)),
 				MinimumBaseAmount:       minBaseAmount,
 				MinimumQuoteAmount:      pairsData[x].MinQuoteAmount.Float64(),
 			})
 		}
+	case asset.Futures:
+		btcContracts, err := g.GetAllFutureContracts(ctx, currency.BTC)
+		if err != nil {
+			return err
+		}
+		usdtContracts, err := g.GetAllFutureContracts(ctx, currency.USDT)
+		if err != nil {
+			return err
+		}
+		btcContracts = append(btcContracts, usdtContracts...)
+		l = make([]limits.MinMaxLevel, 0, len(btcContracts))
+		for x := range btcContracts {
+			if btcContracts[x].InDelisting {
+				continue
+			}
+			p := strings.ToUpper(btcContracts[x].Name)
+			if !g.IsValidPairString(p) {
+				continue
+			}
```

Comment:
Can you please have a look at this function, it seems like it can be removed completely?